packages/upswyng-web/src/components/Resource.tsx

```diff
@@ -81,7 +84,11 @@ export const Resource = () => {
 
   return (
     <Container>
-      <PageBanner color={resourceColor} text={resource.name} />
+      <PageBanner
+        color={resourceColor}
+        text={resource.name}
+        backRef={history.goBack}
```

Comment:
The last issue with this implementation is that jumping directly to an individual resource page will result in the back button taking them to any previous page. For example:

1. open a new tab
2. navigate to https://deploy-preview-270--upswyng.netlify.com/resource/5def5cb7b831a048f2a3deed
3. press the back button

And it will take you back to your new tab page rather than, ideally, Upswyng's home page.

What we need to do is set this set this value to either:

A. value that is passed to the `<Resource/>` component
B. `history.goBack` only when the previous page is within Upswyng

Since `<Resource/>` is render by a route found in the main `<App/> ` passing something directly to that component won't work; however, we could use [React context](https://reactjs.org/docs/context.html) to accomplish this.

I'm not totally in love with this solution, but it's the first thing that comes to mind:

1. create a new or update some existing context that sets a the `backRef` value as `history.push('/')` by default
2. update both `<ResourceList/>` and `<SearchResults/>` to consume that context a update the `backRef` to be `history.goBack()`
3. update `<PageBanner/>` to consume that context and call `backRef` as it is before.

I now remember why I used context for setting the banner color of a resource. ([example of context provider](https://github.com/CodeForBoulder/upswyng/blob/master/packages/upswyng-web/src/components/PageBanner.tsx))

The similar problem I had was that a Resource's banner color was equal to either, in order of importance:
- the color of the category that led to the resource
- the first category color it belonged to when it was navigated to from the search results or directly

The solution I went with was that whenever a banner was given a specific color (in the case of a category/subcategory page), it would [change the color context to match it](https://github.com/CodeForBoulder/upswyng/blob/master/packages/upswyng-web/src/components/PageBanner.tsx#L37). It would then ensure that future page banners would also use that color until it was reset elsewhere, as with the [search page](https://github.com/CodeForBoulder/upswyng/blob/master/packages/upswyng-web/src/components/Search.tsx#L12).

You could create some new context for the `backRef` to accomplish something similar.

@rhinodavid could I get a sanity check?

Reply:
seems insane

Reply:
> seems insane

What would a better option be? The only options I can think of to flag whether the last location was within the app is:

- React context
- some indication in the URL (query param or optional slug?)
- local storage